From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DoctaEnkoda <bierquejason@gmail.com>
Date: Fri, 16 Jul 2021 18:08:12 +0200
Subject: [PATCH] Micro-Optimization bee


diff --git a/src/main/java/net/minecraft/world/entity/animal/EntityBee.java b/src/main/java/net/minecraft/world/entity/animal/EntityBee.java
index 4da4fadbd4a036b316d554de20c0d30d35cc206b..92d218342854ef6ae818a417a985ca59a2fe9431 100644
--- a/src/main/java/net/minecraft/world/entity/animal/EntityBee.java
+++ b/src/main/java/net/minecraft/world/entity/animal/EntityBee.java
@@ -295,6 +295,11 @@ public class EntityBee extends EntityAnimal implements IEntityAngerable, EntityB
     @Override
     public void tick() {
         super.tick();
+        // Purpur-SinonMC Start - We kill the bee directly.
+        if (this.getBlockLocation().getY() <= -1) {
+            this.killEntity();
+        } else
+        // Purpur-SinonMC Stop
         if (this.hasNectar() && this.getNumCropsGrownSincePollination() < 10 && this.random.nextFloat() < 0.05F) {
             for (int i = 0; i < this.random.nextInt(2) + 1; ++i) {
                 this.a(this.world, this.locX() - 0.30000001192092896D, this.locX() + 0.30000001192092896D, this.locZ() - 0.30000001192092896D, this.locZ() + 0.30000001192092896D, this.e(0.5D), Particles.FALLING_NECTAR);
@@ -630,7 +635,7 @@ public class EntityBee extends EntityAnimal implements IEntityAngerable, EntityB
 
     @Override
     protected float b(EntityPose entitypose, EntitySize entitysize) {
-        return this.isBaby() ? entitysize.height * 0.5F : entitysize.height * 0.5F;
+        return entitysize.height * 0.5F; // Purpur-SinonMC Remove useless condition
     }
 
     @Override
