From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bierque Jason <bierquejason@gmail.com>
Date: Fri, 26 Nov 2021 14:13:52 +0100
Subject: [PATCH] Micro Optimization Light Engine


diff --git a/src/main/java/net/minecraft/server/level/ThreadedLevelLightEngine.java b/src/main/java/net/minecraft/server/level/ThreadedLevelLightEngine.java
index 136ca55ac38b709e086f8e2e68dc1d0db2670ae4..9d5b24264fcf25d225e3d7c0e468d2e2f912ea7a 100644
--- a/src/main/java/net/minecraft/server/level/ThreadedLevelLightEngine.java
+++ b/src/main/java/net/minecraft/server/level/ThreadedLevelLightEngine.java
@@ -323,8 +323,10 @@ public class ThreadedLevelLightEngine extends LevelLightEngine implements AutoCl
 
     @Override
     public int getRawBrightness(final BlockPos pos, final int ambientDarkness) {
-        // need to use new light hooks for this
+        if (ambientDarkness == 0) return 15; // Purpur-SinonMC Micro Optimization Light Engine
+        // need to use new light  hooks for this
         final int sky = this.theLightEngine.getSkyReader().getLightValue(pos) - ambientDarkness;
+        // Don't fetch the block light level if the skylight level is 15, since the value will never be higher.
         final int block = this.theLightEngine.getBlockReader().getLightValue(pos);
         return Math.max(sky, block);
     }
